src/scales.js

```diff
@@ -136,20 +136,21 @@ function Scale(key, channels = [], options = {}) {
   const type = inferScaleType(key, channels, options);
 
   // Warn for common misuses of implicit ordinal scales. We disable this test if
-  // you set the domain or range explicitly, since setting the domain or range
-  // (typically with a cardinality of more than two) is another indication that
-  // you intended for the scale to be ordinal; we also disable it for facet
-  // scales since these are always band scales.
+  // you specify a scale interval or if you set the domain or range explicitly,
+  // since setting the domain or range (typically with a cardinality of more than
+  // two) is another indication that you intended for the scale to be ordinal; we
+  // also disable it for facet scales since these are always band scales.
   if (options.type === undefined
       && options.domain === undefined
       && options.range === undefined
+      && options.interval == null
```

Comment:
```suggestion
      && options.interval === undefined
```

Assuming we do maybeInterval on scale construction (per other comments) we don’t have to worry about null vs. undefined here, too.